Updated page "updated firmware" to reflect recent changes in the Server (#341)
SlimeVR/SlimeVR-Docs-Site: src/updating-firmware.md

# Updating Your Tracker's Firmware

> On this page you will find several ways to flash SlimeVR trackers and how to recover official SlimeVR trackers 

Reviewer: `On this page, you will find several methods for flashing SlimeVR trackers and instructions on how to recover official SlimeVR trackers.`

> On this page you will find several ways to flash SlimeVR trackers and how to recover official SlimeVR trackers 

Reviewer: ```suggestion
On this page, you will find several methods for flashing SlimeVR trackers and instructions on how to recover official SlimeVR trackers.
```

## Index

> - [Update Your Slimes](#update-your-slimes) - How to update your official SlimeVR trackers.
> - [USB Recovery](#usb-recovery) - Recover trackers after failed update.
> - [Manual Wireless Updates](#manual-wireless-updates) - For rolling back firmware.

Reviewer: ```suggestion
- [Update Your SlimeVR Trackers](#update-your-slimes) - A Guide to Updating Your Official SlimeVR Trackers
- [USB Recovery](#usb-recovery) - Restore Trackers After a Failed Update
- [Manual Wireless Updates](#manual-wireless-updates) - Rolling Back Firmware
```

# Update Your Slimes

> Official slimes should under normal circumstances be updated using the update icon that appears on top of the tracker in the main menu.

Reviewer: ```suggestion
Official SlimeVR Trackers should, under normal circumstances, be updated using the update icon that appears at the top of the tracker in the main menu.
```

![update](assets/img/Official_Tracker_Update.png)

> Beware that the trackers need to be charged to at least 50% for this to work.

Reviewer: ```suggestion
Please ensure that the trackers are charged to at least 50% for this to function properly.
```

![updatelowbattery](assets/img/Official_Tracker_Update_Low_Battery.png)

 

> Upon clicking the update icon you should be met with the following screen, please select the trackers you wish to update and click "Update Selected Trackers"

Reviewer: ```suggestion
Upon clicking the update icon, you will be presented with the following screen. Please select the trackers you wish to update and click "Update Selected Trackers".
```

![update1](assets/img/Official_Tracker_Update_1.png)

> Now turn the trackers selected for the update off and on again!

Reviewer: ```suggestion
Now, turn the selected trackers for the update off and then back on again!
```

```admonish warning

> Do not unplug or turn off the tracker during the upload process unless told to do so, it may make the tracker unusable.

Reviewer: ```suggestion
Please do not unplug or turn off the tracker during the upload process unless instructed to do so, as this may render the tracker unusable.
```

```

![update2](assets/img/Official_Tracker_Update_2.png)

> Now the software will upload the firmware to the selected trackers.

Reviewer: ```suggestion
Now, the software will upload the firmware to the selected trackers.
```

![update3](assets/img/Official_Tracker_Update_3.png)

> Next the software will apply the updated firmware to the selected trackers.

Reviewer: ```suggestion
Next, the software will apply the updated firmware to the selected trackers.
```

![update4](assets/img/Official_Tracker_Update_4.png)

> And that's it, you should now be set and up to date!

Reviewer: ```suggestion
And that's it! You should now be all set and up to date.
```

![update5](assets/img/Official_Tracker_Update_5.png)

> Now turn the trackers selected for the update off and on again!

Reviewer: ```suggestion
Now, turn the selected trackers for the update off and then back on again!
```

```admonish warning

> If a tracker has stopped working after the update, do not flash any other trackers and refer to the USB recovery section below to recover the trackers firmware.

Reviewer: ```suggestion
If a tracker has stopped working after the update, do not flash any other trackers. Instead, refer to the USB recovery section below to restore the tracker's firmware.
```

```

## Disclaimer
```admonish danger

> These steps linked below are tailored towards official SlimeVR trackers, the settings pictured will not work for most DIY trackers!
> This method has the potential to mess up your tracker, so please read the following warnings:

Reviewer: ```suggestion
These steps linked below are specifically designed for Official SlimeVR Trackers; the settings shown will not be compatible with most DIY trackers.
This method has the potential to soft brick your tracker, so please read the following warnings carefully:
```

> 1. Do not turn off your trackers during the flashing/applying update processes.

Reviewer: ```suggestion
1. Do not turn off your trackers during the flashing or applying update processes.
```

> 2. After flashing, ensure that your tracker is functional before turning it off.

Reviewer: ```suggestion
2. After flashing, verify that your tracker is operational before powering it down.
```

> 3. If a tracker has stopped working after flashing it wirelessly, Please try fixing it first performing a USB recovery before continuing with the other trackers.

Reviewer: ```suggestion
3. If a tracker has stopped working after being flashed wirelessly, please attempt to fix it by performing a USB recovery before proceeding with the other trackers.
```

> 4. Uploading unofficial/unsupported firmware files may affect your trackers functionality and may void your warranty.

Reviewer: ```suggestion
4. Uploading unofficial or unsupported firmware files may affect your tracker's functionality and could void your warranty.
```

```

# USB Recovery

> This method should only be used when OTA is not an option.
> During this step you are expected to disasemble the tracker, **this does not void your warranty.**

Reviewer: ```suggestion
This method should only be used when over-the-air (OTA) updates are not an option. 
During this step, you are expected to disassemble the tracker; **this action does not void your warranty**.
```

```admonish warning
This method requires the following items:
 - Data-capable USB-C cable (included one should suffice).
 - M3 **Pozidrive** screwdriver (JIS and Phillips head may work but can strip out the screws).
 - Metal tweezers, a paperclip or comparable conductive tool to bridge pads (only needed for R11 & R12, R14 has a physical button).
```

> *If your screws are stripping out or about to you can use a rubber band to add more grip.*

Reviewer: ```suggestion
*If your screws are stripping or about to strip, you can use a rubber band to provide additional grip.*
```

## Disassembling The Tracker

> Turn off your tracker, remove the strap then use a **pozidrive** screwdriver (JIS and Phillips head may work but can strip out the screws) to undo the screws on the back of the tracker.

Reviewer: ```suggestion
Please turn off your tracker and remove the strap. Then, use a **Pozidriv** screwdriver (a JIS or Phillips head may work, but they can strip the screws) to unscrew the screws on the back of the tracker.
```

![Tracker Screw Disassembly](assets/img/Tracker_Unscrew.png)
 

> Remove the back cover,the battery then gently lift the PCB out at an angle by lifting the opposite side of where all the ports are.

Reviewer: ```suggestion
Remove the back cover and the battery, then gently lift the PCB out at an angle by lifting the side opposite to where all the ports are located.
```

![Tracker Disassembly](assets/img/Tracker_Disassemble.png)

> Inspect the back of the PCB to determine which revision you have.

Reviewer: ```suggestion
Inspect the back of the PCB to identify the revision you possess.
```

![Tracker PCB Revision](assets/img/Board_Identification.png)

## Flashing Via USB (Serial)

> You can find the "DIY Firmware Tool" under your settings menu in the SlimeVR software.
> 
> Firstly select "SlimeVR" if you are using official trackers and click "Next Step"

Reviewer: ```suggestion
You can locate the "DIY Firmware Tool" in the settings menu of the SlimeVR software.

Firstly, select "SlimeVR" if you are using official trackers, and then click "Next Step".
```

![step1](assets/img/Tracker_Flashing_1.png)

> Select the latest available firmware by selecting the highest number in the list and click "Next Step"

Reviewer: ```suggestion
Select the latest available firmware by choosing the highest version in the list, then click "Next Step".
```

![step2](assets/img/Tracker_Flashing_2.png)

> For this step your trackers need to be on and plugged in.
> To Flash the tracker via USB select "serial", put in your Wi-Fi Credentials and select the detected serial device (should be "USB-SERIAL CH340")

Reviewer: ```suggestion
For this step, ensure that your trackers are powered on and connected. 
To flash the tracker via USB, select "Serial", enter your Wi-Fi credentials, and choose the detected serial device (it should be labeled "USB-SERIAL CH340")
```

```admonish warning

> The Wi-Fi name and password are both case sensitive!

Reviewer: ```suggestion
The Wi-Fi name and password are both case-sensitive.
```

```

![step3](assets/img/Tracker_Flashing_4_SRL.png)

> Now for the next step you will have to boot the tracker into bootloader mode by doing the following.
> You know you will have done it correctly if the blue LED gives a single fast blink when turned on using this method.

Reviewer: ```suggestion
Now, for the next step, you will need to boot the tracker into bootloader mode by following these instructions.
You will know you have done it correctly if the blue LED emits a single fast blink when powered on using this method.
```

```admonish warning

> Check the back of your tracker to see what revision you have!

Reviewer: ```suggestion
Check the back of your tracker to determine which revision you have!
```

```

|  Revision |                                                                             Steps                                                                           |
|-----------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|

> |    R11    | Turn on the tracker whilst shorting the second rectangular FLASH pad from the edge on the top side of the board to the metal shield of the microcontroller. |
> |    R12    |            Turn on the tracker whilst shorting the circular FLASH pad on the top side of the board, and the metal shield of the microcontroller.               |
> |    R14    |                                           Turn on the tracker while holding the FLASH button on the top side of the board, you can let go after turning it on.                           |

Reviewer: ```suggestion
|    R11    | Turn on the tracker whilst shorting the second rectangular FLASH pad from the edge on the top side of the board to the metal shield of the microcontroller. |
|    R12    | Turn on the tracker while shorting the circular FLASH pad on the top side of the board and the metal shield of the microcontroller. |
|    R14    | Turn on the tracker, press and hold the FLASH button located on the top side of the board. You can release the button after the tracker is turned on. |
```

![Tracker R11](assets/img/R11_board_reset.png)
![Tracker R12](assets/img/R12_board_reset.png)
![Tracker R14](assets/img/R14_board_reset_sw.png)

> Now hit "Next Step" and it should move on to the following screen where it will synchronize with the MCU.
> If this fails you might have to click retry and/or retry the above step to ensure it's in bootloader mode (The Blue LED should give a single fast blue blink when powered on into bootloader mode).

Reviewer: ```suggestion
Now click "Next Step" and the process should advance to the next screen, where it will synchronize with the MCU.
If this fails, you may need to click "Retry" and/or "Retry" from the previous step to ensure that the device is in download mode. The blue LED should emit a single fast blue blink when powered on in download mode.
```

![step8](assets/img/Tracker_Flashing_8.png)

> Now the tracker will begin applying the update.

Reviewer: ```suggestion
The tracker will now begin applying the update.
```

```admonish danger

> Do not turn off the tracker while it is applying the update!

Reviewer: ```suggestion
Please do not turn off the tracker while the update is being applied!
```

```

![step9](assets/img/Tracker_Flashing_9.png)

> After this the update should be complete!
> If you get an error you can click retry, if that doesn't work you may have to start over.

Reviewer: ```suggestion
After this, the update should be complete! If you encounter an error, you can click "Retry".
If that doesn't resolve the issue, you may need to start over.
```

![step10](assets/img/Tracker_Flashing_10.png)

> Your tracker should now be updated, turn it off and on again to get it connect to Wi-Fi again, you can repeat the process for any other trackers you wish to update.

Reviewer: ```suggestion
Your tracker should now be updated. Please turn it off and then back on to reconnect to Wi-Fi. You can repeat this process for any other trackers you wish to update.
```

# Manual Wireless Updates

> This method is the simplest, and doesn't require any cables or disassembly, but it does require your tracker to already be paired to your Wi-Fi network.
> You can find the "DIY Firmware Tool" under your settings menu in the SlimeVR software.
> 
> Firstly select "SlimeVR" if you are using official trackers and click "Next Step"

Reviewer: ```suggestion
This method is the simplest and does not require any cables or disassembly; however, it does necessitate that your tracker is already paired with your Wi-Fi network.
You can find the "DIY Firmware Tool" in the Settings menu of the SlimeVR software.

Firstly, select "SlimeVR" if you are using official trackers, and then click "Next Step".
```

![step1](assets/img/Tracker_Flashing_1.png)

> Select the latest available or desired firmware by selecting the version number in the list and click "Next Step"

Reviewer: ```suggestion
Select the latest available or desired firmware by choosing the version number from the list, then click "Next Step".
```

![step2](assets/img/Tracker_Flashing_2.png)

> To Flash the tracker wirelessly select "OTA" and then select all the trackers you wish to flash.
> For this step your trackers need to be on and connected to your Wi-Fi, once selected click "Next Step"

Reviewer: ```suggestion
To flash the tracker wirelessly, select "OTA" and then choose all the trackers you wish to update.
For this step, your trackers must be powered on and connected to your Wi-Fi. Once you have made your selections, click "Next Step".
```

![step3](assets/img/Tracker_Flashing_3_OTA.png)

> On this step you will have to turn the tracker(s) off and back on so we can verify OTA is enabled and start flashing them.

Reviewer: ```suggestion
On this step, you will need to turn the tracker(s) off and then back on to verify that OTA is enabled and to begin flashing them.
```

```admonish warning

> If you get an error, and the tool says "Timeout", you need to click "Retry" and follow the instructions.

Reviewer: ```suggestion
If you encounter an error and the tool displays "Timeout" click "Retry" and follow the provided instructions.
```

```

![step5](assets/img/Tracker_Flashing_5.png)

The tracker(s) will now start updating.
```admonish danger
Do not turn off the tracker while it is flashing!
```

![step6](assets/img/Tracker_Flashing_6.png)

> The tracker is now updated and ready to go!

Reviewer: ```suggestion
The tracker(s) will now begin updating.
```

![step7](assets/img/Tracker_Flashing_7.png)

```admonish danger

> If a tracker has stopped working after flashing it, do not flash any other trackers and refer to the USB flashing section below for recovery.

Reviewer: ```suggestion
If a tracker has stopped functioning after being flashed, do not flash any additional trackers. Instead, refer to the USB flashing section below for recovery instructions.
```
